lib/msf/core/auxiliary/web.rb

```diff
@@ -253,13 +253,13 @@ def process_vulnerability( element, proof, opts = {} )
 
     confidence = calculate_confidence( parent.vulns[mode][vhash] )
 
-    parent.vulns[mode][vhash].merge!( :confidence => confidence )
+    parent.vulns[mode][vhash][:confidence] = confidence
 
     if !(payload = opts[:payload])
       if payloads
         payload = payloads.select { |p|
           element.altered_value.include?( p )
-        }.sort_by { |p| p.size }.last
+        }.max
```

Comment:
Pretty sure this will need to be `max_by &:size`, since we don't specify `<=>` for payloads IIRC.